src/persistence/SubjectDatabaseEntityLoader.ts

```diff
@@ -123,20 +124,17 @@ export class SubjectDatabaseEntityLoader {
                         .getMany()
                 }
 
-                // now when we have entities we need to find subject of each entity
-                // and insert that entity into database entity of the found subjects
+                // Now when we have entities we need to find subject of each entity
+                // and insert that entity into database entity of the found subjects.
+                // A single entity can be applied to many subjects as there might be duplicates.
+                // This will likely result in the same row being updated multiple times during a transaction.
                 entities.forEach((entity) => {
-                    const subjects = this.findByPersistEntityLike(
-                        subjectGroup.target,
-                        entity,
-                    )
-                    subjects.forEach((subject) => {
-                        subject.databaseEntity = entity
-                        if (!subject.identifier)
-                            subject.identifier =
-                                subject.metadata.hasAllPrimaryKeys(entity)
-                                    ? subject.metadata.getEntityIdMap(entity)
-                                    : undefined
+                    const entityId =
+                        allSubjects[0].metadata.getEntityIdMap(entity)
```

Comment:
I feel like I'm missing something about how this works. Isn't this taking the id map from the first subject and using it for comparisons against every other subject? I would expect that to only match against the first subject and silently ignore the rest.

Do we have tests to verify the expected output is identical here?

Reply:
A similar way to get the same `EntityMetadata` is a few lines above.

Reply:
Ah I see - we're not getting the id of the first subject, we're using it's method to get the id for `entity`